src/client/components/AreaSelector/CountryList/CountryList.tsx

```diff
@@ -100,32 +86,36 @@ const CountryList: React.FC<Props> = (props: Props) => {
             })}
         </div>
 
-        {includeCountries && showCountryRole
-          ? Object.entries(countryMap).map(([role, cycleCountries]) => (
-              <CountryListRoleSection
-                key={role}
-                countryISOs={cycleCountries[cycle.uuid]}
+        {showCountriesWithRoles &&
+          Object.entries(countryMap).map(([role, cycleCountries]) => (
+            <CountryListRoleSection
+              key={role}
+              countryISOs={cycleCountries[cycle.uuid]}
+              onElementSelect={onElementSelect}
+              query={query}
+              role={role}
+              selectedValue={selectedValue}
+            />
+          ))}
+
+        {showCountriesWithoutRoles &&
```

Comment:
Introduce separate booleans for showing rows with or without role.
Fixes issue where all countries were displayed and makes code a bit more readable.

See issue for screenshot